x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/amazonbedrock/client/AmazonBedrockStreamingProcessor.java

```diff
@@ -7,52 +7,35 @@
 
 package org.elasticsearch.xpack.inference.services.amazonbedrock.client;
 
-import software.amazon.awssdk.services.bedrockruntime.model.ContentBlockDeltaEvent;
 import software.amazon.awssdk.services.bedrockruntime.model.ConverseStreamOutput;
-import software.amazon.awssdk.services.bedrockruntime.model.ConverseStreamResponseHandler;
 
 import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.ExceptionsHelper;
+import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.util.concurrent.EsExecutors;
-import org.elasticsearch.core.Strings;
 import org.elasticsearch.logging.LogManager;
 import org.elasticsearch.logging.Logger;
 import org.elasticsearch.threadpool.ThreadPool;
-import org.elasticsearch.xpack.core.inference.results.StreamingChatCompletionResults;
 
-import java.util.ArrayDeque;
 import java.util.concurrent.Flow;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.atomic.AtomicReference;
 
 import static org.elasticsearch.xpack.inference.InferencePlugin.UTILITY_THREAD_POOL_NAME;
 
-class AmazonBedrockStreamingChatProcessor implements Flow.Processor<ConverseStreamOutput, StreamingChatCompletionResults.Results> {
-    private static final Logger logger = LogManager.getLogger(AmazonBedrockStreamingChatProcessor.class);
+abstract class AmazonBedrockStreamingProcessor<T> implements Flow.Processor<ConverseStreamOutput, T> {
+    private static final Logger logger = LogManager.getLogger(AmazonBedrockStreamingProcessor.class);
 
-    private final AtomicReference<Throwable> error = new AtomicReference<>(null);
-    private final AtomicLong demand = new AtomicLong(0);
-    private final AtomicBoolean isDone = new AtomicBoolean(false);
-    private final AtomicBoolean onCompleteCalled = new AtomicBoolean(false);
-    private final AtomicBoolean onErrorCalled = new AtomicBoolean(false);
-    private final ThreadPool threadPool;
-    private volatile Flow.Subscriber<? super StreamingChatCompletionResults.Results> downstream;
-    private volatile Flow.Subscription upstream;
+    final AtomicReference<Throwable> error = new AtomicReference<>(null);
+    final AtomicLong demand = new AtomicLong(0);
+    final AtomicBoolean isDone = new AtomicBoolean(false);
+    final AtomicBoolean onCompleteCalled = new AtomicBoolean(false);
+    final AtomicBoolean onErrorCalled = new AtomicBoolean(false);
+    final ThreadPool threadPool;
```

Comment:
`error`, `onErrorCalled` and `threadPool` can all remain `private` since they're not used by the implementing classes.